daemon/src/db.rs

```diff
@@ -241,6 +241,82 @@ async fn load_latest_cfd_state(
     Ok(latest_cfd_state_in_db)
 }
 
+pub async fn load_cfd_by_order_id(
+    order_id: OrderId,
+    conn: &mut PoolConnection<Sqlite>,
+) -> anyhow::Result<Cfd> {
+    let order_uuid = serde_json::to_string(&order_id)?;
+
+    let row = sqlx::query!(
+        r#"
+        select
+            cfds.id as cfd_id,
+            orders.uuid as order_id,
+            orders.initial_price as price,
+            orders.min_quantity as min_quantity,
+            orders.max_quantity as max_quantity,
+            orders.leverage as leverage,
+            orders.trading_pair as trading_pair,
+            orders.position as position,
+            orders.origin as origin,
+            orders.liquidation_price as liquidation_price,
+            orders.creation_timestamp as creation_timestamp,
+            orders.term as term,
+            cfds.quantity_usd as quantity_usd,
+            cfd_states.state as state
+        from cfds as cfds
+        inner join orders as orders on cfds.order_uuid = ?
+        inner join cfd_states as cfd_states on cfd_states.cfd_id = cfds.id
+        where cfd_states.state in (
+            select
+              state
+              from cfd_states
+            where cfd_id = cfds.id
+            order by id desc
+            limit 1
+        )
+        "#,
+        order_uuid
+    )
+    .fetch_one(conn)
+    .await?;
+
+    let order_id = serde_json::from_str(row.order_id.as_str()).unwrap();
```

Comment:
We may want to implement `Encode` & `Decode` for some of these types to remove these conversions.

https://docs.rs/sqlx/0.5.7/sqlx/trait.Decode.html

There is also a custom-derive that one can use! https://docs.rs/sqlx/0.5.7/sqlx/derive.Decode.html

Reply:
If all fields implement decode, we may even be able to use `query_as` directly!

Reply:
I recorded this one in a refactoring ticket: https://github.com/comit-network/hermes/issues/88
I did not add this ticket to any epic so far as I don't think it's critical. We could still add it to the project though and just do it at some point.